app/controllers/links_controller.rb

```diff
@@ -0,0 +1,23 @@
+class LinksController < ApplicationController
+  before_action :ensure_current_user
+
+  def destroy
+    link = Link.find(link_params[:id])
+    url = link.url
+    paper = link.paper.title
+    link.destroy
+
+    reference = Reference.find(params[:reference])
+    respond_to do |format|
+      format.html { redirect_back(fallback_location: root_path, notice: "'#{url}' has been successfully removed from '#{paper}'") }
+      format.js { render('destroy_link.js.erb', locals: { reference: reference }) }
```

Comment:
Since `destroy_link.js.erb` is in the `links` folder and is being referred to as its own resource, the common convention would just be to name this `destroy.js.erb`